client-misk-jooq/src/main/kotlin/app/cash/backfila/client/misk/jooq/internal/BatchRangeIterator.kt

```diff
@@ -0,0 +1,76 @@
+package app.cash.backfila.client.misk.jooq.internal
+
+import app.cash.backfila.client.misk.jooq.JooqBackfill
+import app.cash.backfila.protos.clientservice.GetNextBatchRangeRequest
+import app.cash.backfila.protos.clientservice.GetNextBatchRangeResponse
+import com.google.common.base.Stopwatch
+import com.google.common.collect.AbstractIterator
+import org.jooq.DSLContext
+import java.lang.IllegalStateException
+import java.util.concurrent.TimeUnit
+import java.util.function.Supplier
+
+/**
+ * Helper class that helps us iterate over batch ranges.
+ */
+class BatchRangeIterator<K>(
+  private val jooqBackfill: JooqBackfill<K, *>,
+  private val session: DSLContext,
+  private val request: GetNextBatchRangeRequest
+) : AbstractIterator<GetNextBatchRangeResponse.Batch>() {
+
+  private val timeElapsed: Supplier<Boolean>
+  private var nextKeyRange: OpenKeyRange<K>
+  init {
+    timeElapsed = if (request.compute_time_limit_ms == null) Supplier { false } else timer(
```

Comment:
awkward formatting

Reply:
this seems a lot harder to parse than just having a timeElapsed() method

Reply:
Copy pasta from java. I need to take a closer look at this. I think it's tracking time across batches? Not sure yet.

Reply:
Fixed using kotlin idiomatic code.